app/containers/AddWidget.ts

```diff
@@ -1,16 +1,17 @@
 // @ts-nocheck File not migrated fully to TS
-import { connect } from 'react-redux';
 import type { ComponentProps, ComponentType } from 'react';
-import { checkIfWidgetIsUsed, installWidget, uninstallWidget, updateWidgetDefinition } from '../actions/widgets';
+import { connect } from 'react-redux';
+import { checkIfWidgetIsUsed, installWidget, replaceWidget, uninstallWidget } from '../actions/widgets';
 import AddWidgetModal from '../components/AddWidgetModal';
-import stageUtils from '../utils/stageUtils';
 import Consts from '../utils/consts';
+import stageUtils from '../utils/stageUtils';
 
 const mapStateToProps = (state, ownProps) => {
     const widgetDefinitions = state.widgetDefinitions.filter(definition => {
         return (
-            stageUtils.isUserAuthorized(definition.permission, state.manager) &&
-            stageUtils.isWidgetPermitted(definition.supportedEditions, state.manager)
+            !definition.loaded ||
+            (stageUtils.isUserAuthorized(definition.permission, state.manager) &&
+                stageUtils.isWidgetPermitted(definition.supportedEditions, state.manager))
```

Comment:
This condition seems to be a little cluttered
What would you think about the suggestion below? 🤔 

```suggestion
        const isLoadingDefinition = !definition.loaded;
        const isUserAuthorized = stageUtils.isUserAuthorized(definition.permission, state.manager);
        const isWidgetPermitted = stageUtils.isUserAuthorized(definition.permission, state.manager);

        return isLoadingDefinition || (isUserAuthorized && isWidgetPermitted);
```

Reply:
It would introduce a bug :slightly_smiling_face:. But I applied the idea.